File: saveconfiguration.m
%% This routine prints the particle-tracking experiment configuration to a
% text file. The filename includes the date and time of the simulation.

disp('SAVE EXPERIMENT CONFIGURATION... ')

% Get current time
nowtime = clock;
nowtime = datestr(nowtime,'ddmmmyy-HHMM');

% Open the file
fileID = fopen([particle.outputdir,'/',nowtime,'particle_tracking_configuration','.out'],'w');

% Write the data
fprintf(fileID,'%s\n','******** OCEAN MODEL PARAMETERS ********');
fprintf(fileID,'%s\n',['Start day = ',num2str(model.start_day)]);
fprintf(fileID,'%s\n',['Ocean model timestep = ',num2str(model.timestep),' seconds']);
fprintf(fileID,'%s\n',['Directory of ocean model outputs = ',model.path]);
fprintf(fileID,'%s\n',['Ocean model resolution used = every ',num2str(model.outputskip),' file(s)']);
fprintf(fileID,'%s\n',['Ocean model periodicities: E-W = ',num2str(model.periodic_ew),', N-S = ',num2str(model.periodic_ns)]);
fprintf(fileID,'%s\n',[]);
fprintf(fileID,'%s\n',[]);
fprintf(fileID,'%s\n','******** PARTICLE TRACKING PARAMETERS ********');
fprintf(fileID,'%s\n',['Particle initialization (1st seeding) = Day ',num2str(particle.initime)]);
fprintf(fileID,'%s\n',['Particle seeding frequency = every ',num2str(particle.inifreq),' day(s)']);
fprintf(fileID,'%s\n',['Timestep for particle tracking = ',num2str(particle.timestep),' day(s)']);
fprintf(fileID,'%s\n',['Length of particle tracking simulation = ',num2str(particle.length),' day(s)']);
fprintf(fileID,'%s\n',['Particle output frequency = ',num2str(particle.outfreq),' day(s)']);
fprintf(fileID,'%s\n',['Particle output format = ',particle.outputformat]);
fprintf(fileID,'%s\n',['Particle output directory = ',particle.outputdir]);
fprintf(fileID,'%s\n',['Particle output filename = ',particle.outputfilename]);
fprintf(fileID,'%s\n',[]);
fprintf(fileID,'%s\n',['Particle tracking direction: ',particle.direction]);
fprintf(fileID,'%s\n',[]);
fprintf(fileID,'%s\n','Particle seeding strategy in x:');
fprintf(fileID,'%s\n',['istart = ',num2str(particle.istart)]);
fprintf(fileID,'%s\n',['irange = ',num2str(particle.irange)]);
fprintf(fileID,'%s\n',['irez = ',num2str(particle.irez)]);
fprintf(fileID,'%s\n','Particle seeding strategy in y:');
fprintf(fileID,'%s\n',['jstart = ',num2str(particle.jstart)]);
fprintf(fileID,'%s\n',['jrange = ',num2str(particle.jrange)]);
fprintf(fileID,'%s\n',['jrez = ',num2str(particle.jrez)]);
fprintf(fileID,'%s\n','Particle seeding strategy in z:');
fprintf(fileID,'%s\n',['kstart = ',num2str(particle.kstart)]);
fprintf(fileID,'%s\n',['krange = ',num2str(particle.krange)]);
fprintf(fileID,'%s\n',['krez = ',num2str(particle.krez)]);
fprintf(fileID,'%s\n',['Number of particle classes = ',num2str(particle.numofclasses)]);

% Close the file
fclose(fileID);
clear fileID nowtime ans

disp('DONE')
disp('%%%%%%%%%%%%%%%%%')
disp(' ')

File: python/saveconfiguration.py
## This routine prints the particle-tracking experiment configuration to a
# text file. The filename includes the date and time of the simulation.
def saveconfiguration(model,particle):
    print('SAVE EXPERIMENT CONFIGURATION... ')
    import datetime

    # Get current time with datetime
    nowtime = datetime.datetime.now().strftime(' %d%m%y-%H%M')

    # Open the file
    file = open('%s/%s_%s_configuration.out' %(particle.outputdir,nowtime,particle.outputfilename),'w')

    # Write the data
    file.write('******** OCEAN MODEL PARAMETERS ********\n')
    file.write('Start day =  %f \n' %model['start_day'])
    file.write('Ocean model timestep =  %f seconds \n' %model['timestep'])
    file.write('Directory of ocean model outputs = %s \n' %model['path'])
    file.write('Ocean model resolution used = every  %f file(s) \n' %model['outputskip'])
    file.write('Ocean model periodicities: E-W =  %f  N-S =  %f \n' %(model['periodic_ew'],model['periodic_ns']))
    file.write('\n')
    file.write('\n')
    file.write('******** PARTICLE TRACKING PARAMETERS ********\n')
    file.write('\n')
    file.write('#----------------\n')
    file.write('SEEDING\n')
    file.write('#----------------\n')
    file.write('Particle initialization (1st seeding) = Day  %f \n' %particle.initime)
    file.write('Particle seeding frequency = every  %f day(s) \n' %particle.inifreq)
    file.write('Number of seeding events =  %f \n' %particle.ininumber)
    file.write('Type of seeding (static vs dynamic) = %s \n' %particle.initype)
    file.write('\n')
    file.write('Number of particle classes =  %f \n' %particle.numofclasses)
    file.write('\n')
    file.write('Particle seeding strategy in x:\n')
    file.write('istart =  %f \n' %particle.istart)
    file.write('irange =  %f \n' %particle.irange)
    file.write('irez =  %f \n' %particle.irez)
    file.write('Particle seeding strategy in y:\n')
    file.write('jstart =  %f \n' %particle.jstart)
    file.write('jrange =  %f \n' %particle.jrange)
    file.write('jrez =  %f \n' %particle.jrez)
    file.write('Particle seeding strategy in z:\n')
    file.write('kstart =  %f \n' %particle.kstart)
    file.write('krange =  %f \n' %particle.krange)
    file.write('krez =  %f \n' %particle.krez)
    file.write('\n')
    file.write('#----------------\n')
    file.write('ADVECTION\n')
    file.write('#----------------\n')
    file.write('Timestep for particle tracking =  %f day(s)\n' %particle.timestep)
    file.write('Length of particle tracking simulation =  %f day(s)\n' %particle.length)
    file.write('Particle tracking direction: %s \n' %particle.direction)
    file.write('\n')
    file.write('#----------------\n')
    file.write('OUTPUT\n')
    file.write('#----------------\n')
    file.write('Particle output frequency =  %f day(s)\n' %particle.outfreq)
    file.write('Particle output format = %s \n' %particle.outputformat)
    file.write('Particle output directory =%s \n' %particle.outputdir)
    file.write('Particle output filename = %s \n' %particle.outputfilename)

    # Close the file
    file.close()

    print('DONE\n')
    print('#################\n')
    print('\n')
    return